website/project/model.py

```diff
@@ -852,16 +856,48 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
     # Primary institution node is attached to
     primary_institution = fields.ForeignField('institution')
 
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
     def add_primary_institution(self, user, inst):
-        if inst.auth(user):
-            self.primary_institution = inst
-            self.save()
-            return True
-        return False
+        if not user.is_affiliated_with_institution(inst):
+            raise UserNotAffiliatedError('User is not affiliated with {}'.format(inst.name))
+        if inst == self.primary_institution:
+            return False
+        self.primary_institution = inst
+        if inst not in self.affiliated_institutions:
+            self.affiliated_institutions.append(inst)
+        self.add_log(
+            action=NodeLog.PRIMARY_INSTITUTION_ADDED,
+            params={
+                'node': self._primary_key,
+                'institution': {
```

Comment:
Just in case we decide to display logs like "change primary institution from X to Y", it would be good to capture the previous institution value (or `None` if there was no previous institution. 

If this is a quick change, go ahead and make it. Otherwise, this can go in a follow-up PR.